WordPress/src/main/java/org/wordpress/android/ui/posts/PostsPagerAdapter.kt

```diff
@@ -11,13 +12,37 @@ class PostsPagerAdapter(
     private val pages: List<PostListType>,
     private val site: SiteModel,
     val fm: FragmentManager
-) : FragmentPagerAdapter(fm) {
+) : FragmentStatePagerAdapter(fm) {
```

Comment:
`FragmentStatePagerAdapter` will actually replace the current fragments if they need to be replaced, where `FragmentPagerAdapter` will keep the fragment instance and recreate the UI.

The high level difference can be represented as "Does the fragment instance represent state?" in our case the answer is yes.

This all said since we will now just replace the Adapter we should be able to keep it as `FragmentPagerAdapter`.

Reply:
Looks like even with replacing the Adapter the fragments don't get switched out correctly unless we are using `FragmentStatePagerAdapter`

Reply:
I'll look into this tomorrow or on Monday. I think the problem is the `FragmentPager` with `FragmentPagerAdapter` keeps certain number of the pages in a cache. I'm not sure if there is a way how to clear the cache after we replace the adapter.

Note: I also always thought the `FragmentPagerAdapter` should be used for low number of pages (keeps certain number of the fragments in memory) and `FragmentStatePagerAdapter` for high number of pages (eg. e-book reader which doesn't need to keep the fragments in memory).

Reply:
@malinajirka I've done a bit of research and found a good article https://medium.com/inloopx/adventures-with-fragmentstatepageradapter-4f56a643f8e0

I think what we are seeing is `FragmentPagerAdapter` using `getItemId` to find the fragment in the FragmentManager based on its position.

I haven't found any good details about whether fragments from old adapters actually get removed from the `FragmentManager` when we switch the adapter using the `FragmentStatePagerAdapter` though.

I'll make the other changes then dig into what is going on afterwards. I think if the `FragmentStatePagerAdapter` is handling removing the old fragments for us then it would be the better choice.

Also worth noting is that @oguzkocer has raised the [question](https://github.com/wordpress-mobile/WordPress-Android/pull/9446#discussion_r267979148) regarding whether we need to change out the fragments instead of just updating the items themselves.

Reply:
>I'll make the other changes then dig into what is going on afterwards. I think if the FragmentStatePagerAdapter is handling removing the old fragments for us then it would be the better choice.

I'm usually not concerned about performance until we have a performance issue (don't quote me on this :D - it really depends on the situation). But in this case I feel like we'd be using a component (`FragmentStatePagerAdapter`) which is meant for another purpose even though a component build exactly for our purpose exists (`FragmentPagerAdapter`).

Since your solution works I'd suggest leaving this for another PR. There is no need to solve everything before we merge this since we are not merging it into `develop` but to a working branch. 

Tbh I wanted to keep things simple so that was one of the reasons I suggested switching the whole adapter instead of recreating the fragments within the same adapter. I'm sorry, I haven't realized it'd introduce this issue. I agree with @oguzkocer that the ideal solution is to just inform the fragments about the filter change - they should take care of updating the items in the list (it'll require some changes in PostListViewModel though).

Reply:
OK cool, will resolve this issue for now then, and we can look at it further in another PR